_tools/linkcheck.ts

```diff
@@ -0,0 +1,36 @@
+import { platform, exit } from 'process';
+import { spawn } from 'child_process';
+
+const server = spawn('yarn', ['serve'], {
+  stdio: ['ignore', 'pipe', 'ignore'],
+});
+
+let output = '';
+server.stdout.on('data', (data) => {
+  console.log(data.toString());
+  output = output + data.toString();
+  if (output.indexOf('Serving files from') !== -1) {
+    let program;
+    if (platform === 'linux') {
+      program = 'linkcheck-linux';
+    } else if (platform === 'darwin') {
+      program = 'linkcheck';
+    } else {
+      program = 'linkcheck-win';
+    }
+    const checker = spawn(program, [':4200'], {
+      stdio: ['ignore', 'pipe', 'ignore'],
+    });
+    checker.stdout.on('data', (checkerData) => {
+      console.log(checkerData.toString());
+    });
+    checker.on('exit', (code, other) => {
+      console.log('linkcheck finished');
+      server.stdout.destroy();
+      server.kill();
+      checker.stdout.destroy();
+      checker.kill();
+      exit(code);
+    });
+  }
```

Comment:
If the `yarn serve` fails, we'd have bigger issues not related to this check (likely caught by the prior `yarn build` test first).  

In a worst case scenario where it fails, it would exit and the CI would complete. If it works but never prints this message, CI would timeout, but there's no way for us to know if it's still starting up, or if it never printed the message (halting problem almost).